DQM/SiPixelMonitorTrack/src/SiPixelTrackResidualSource.cc

```diff
@@ -1586,4 +1574,74 @@ void SiPixelTrackResidualSource::triplets(double x1,double y1,double z1,double x
 }
 
 
+void SiPixelTrackResidualSource::getepixrococcupancyontrk(const TrackerTopology* const tTopo, TransientTrackingRecHit::ConstRecHitPointer hit, 
+							  float xclust, float yclust, float z, MonitorElement* meinput) {
+  
+  int pxfpanel     = tTopo->pxfPanel((*hit).geographicalId());
+  int pxfmodule    = tTopo->pxfModule((*hit).geographicalId());
+  int pxfdisk      = tTopo->pxfDisk((*hit).geographicalId());
+  int pxfblade_off = tTopo->pxfBlade((*hit).geographicalId());
+  
+  // translate to online conventions
+  if (z<0.) { pxfdisk  = -1.*pxfdisk; }
+  int pxfblade = -99;
+  if (pxfblade_off<=6 && pxfblade_off>=1)        { pxfblade = 7-pxfblade_off;  }
+  else if (pxfblade_off<=18 && pxfblade_off>=7)  { pxfblade = 6-pxfblade_off;  }
+  else if (pxfblade_off<=24 && pxfblade_off>=19) { pxfblade = 31-pxfblade_off; }
+ 
+  int clu_sdpx = ((pxfdisk>0) ? 1 : -1) * (2 * (abs(pxfdisk) - 1) + pxfpanel);
+  int binselx = (pxfpanel==1&&(pxfmodule==1||pxfmodule==4)) ? (pxfmodule==1) : ((pxfpanel==1&& xclust<80.0)||(pxfpanel==2&&xclust>=80.0));
+  int nperpan = 2 * pxfmodule + pxfpanel - 1 + binselx;
+  int clu_roc_binx = ((pxfdisk>0) ? nperpan : 9 - nperpan) + (clu_sdpx + 4) * 8 - 2 * ((abs(pxfdisk)==1) ? pxfdisk : 0);
+  
+  int clu_roc_biny = -99.;
+  int nrocly = pxfmodule + pxfpanel;
+  for (int i=0; i<nrocly; i++) {
+    int j = (pxfdisk<0) ? i : nrocly - 1 - i;
+    if (yclust>=(j*52.0)&& yclust<((j+1)*52.0))
```

Comment:
@davidlange6  thanks. I will work on your comments and update the PR later today